dbms/src/Storages/DeltaMerge/tests/gtest_local_index_info.cpp

```diff
@@ -0,0 +1,290 @@
+    {
+        expect_idx.id = 1;
+        expect_idx.idx_cols.emplace_back(default_index_col_info);
+        expect_idx.vector_index = TiDB::VectorIndexDefinitionPtr(new TiDB::VectorIndexDefinition{
+            .kind = tipb::VectorIndexKind::HNSW,
+            .dimension = 1,
+            .distance_metric = tipb::VectorDistanceMetric::L2,
+        });
+        table_info.index_infos.emplace_back(expect_idx);
+    }
+
+    // check the different
+    {
+        auto new_index_info = generateLocalIndexInfos(index_info, table_info, logger);
+        ASSERT_NE(new_index_info, nullptr);
+        ASSERT_EQ(new_index_info->size(), 1);
+        const auto & idx = (*new_index_info)[0];
+        ASSERT_EQ(DM::IndexType::Vector, idx.type);
+        ASSERT_EQ(expect_idx.id, idx.index_id);
+        ASSERT_EQ(100, idx.column_id);
+        ASSERT_NE(nullptr, idx.index_definition);
+        ASSERT_EQ(expect_idx.vector_index->kind, idx.index_definition->kind);
+        ASSERT_EQ(expect_idx.vector_index->dimension, idx.index_definition->dimension);
+        ASSERT_EQ(expect_idx.vector_index->distance_metric, idx.index_definition->distance_metric);
+
+        // check again, nothing changed, return nullptr
+        ASSERT_EQ(nullptr, generateLocalIndexInfos(new_index_info, table_info, logger));
+
+        // update
+        index_info = new_index_info;
+    }
+
+    // Add another vector index to the TableInfo.
+    TiDB::IndexInfo expect_idx2;
+    {
+        expect_idx2.id = 2; // another index_id
+        expect_idx2.idx_cols.emplace_back(default_index_col_info);
+        expect_idx2.vector_index = TiDB::VectorIndexDefinitionPtr(new TiDB::VectorIndexDefinition{
+            .kind = tipb::VectorIndexKind::HNSW,
+            .dimension = 2,
+            .distance_metric = tipb::VectorDistanceMetric::COSINE, // another distance
+        });
+        table_info.index_infos.emplace_back(expect_idx2);
+    }
+    // check the different
+    {
+        auto new_index_info = generateLocalIndexInfos(index_info, table_info, logger);
+        ASSERT_NE(new_index_info, nullptr);
+        ASSERT_EQ(new_index_info->size(), 2);
+        const auto & idx0 = (*new_index_info)[0];
+        ASSERT_EQ(DM::IndexType::Vector, idx0.type);
+        ASSERT_EQ(expect_idx.id, idx0.index_id);
+        ASSERT_EQ(100, idx0.column_id);
+        ASSERT_NE(nullptr, idx0.index_definition);
+        ASSERT_EQ(expect_idx.vector_index->kind, idx0.index_definition->kind);
+        ASSERT_EQ(expect_idx.vector_index->dimension, idx0.index_definition->dimension);
+        ASSERT_EQ(expect_idx.vector_index->distance_metric, idx0.index_definition->distance_metric);
+        const auto & idx1 = (*new_index_info)[1];
+        ASSERT_EQ(DM::IndexType::Vector, idx1.type);
+        ASSERT_EQ(expect_idx2.id, idx1.index_id);
+        ASSERT_EQ(100, idx1.column_id);
+        ASSERT_NE(nullptr, idx1.index_definition);
+        ASSERT_EQ(expect_idx2.vector_index->kind, idx1.index_definition->kind);
```

Comment:
In open-source TiDB, we remove the field of kind. We use the `indexinfo.Tp` instead of this kind of information. link: https://github.com/pingcap/tidb/pull/55839/files#diff-645e83502d90c4de6385256f7499dcdc234ef8b6808559e19570eb2d05c25992R54